src/src/Helpers/Networking.cpp

```diff
@@ -1354,23 +1354,24 @@ int http_authenticate(const String& logIdentifier,
 {
   if (!uri.startsWith(F("/"))) {
     return http_authenticate(
-      logIdentifier, 
-      client, 
-      http, 
-      timeout, 
-      user, 
-      pass, 
-      host, 
-      port, 
+      logIdentifier,
+      client,
+      http,
+      timeout,
+      user,
+      pass,
+      host,
+      port,
       concat(F("/"), uri),
-      HttpMethod, 
-      header, 
+      HttpMethod,
+      header,
       postStr,
       must_check_reply);
   }
   int httpCode = 0;
+  const bool hasCredentials = !user.isEmpty() && !pass.isEmpty();
 
-  if (user.length() && pass.length()) {
+  if (hasCredentials) {
```

Comment:
I'm wondering if we should support the case where an external entity provides a base64 hash to log in, that _might_ not contain a colon. The hash is then usually stored in the password field, and should be passed as the single value to `http.setAuthorization(pass.c_str())`.